llvm/lib/Target/NVPTX/NVPTXISelLowering.cpp

```diff
@@ -6100,6 +6100,8 @@ NVPTXTargetLowering::shouldExpandAtomicRMWInIR(AtomicRMWInst *AI) const {
 
   if (AI->isFloatingPointOperation()) {
     if (AI->getOperation() == AtomicRMWInst::BinOp::FAdd) {
+      if (Ty->isHalfTy() && STI.getSmVersion() >= 70)
+        return AtomicExpansionKind::None;
```

Comment:
It also requires PTX ISA 6.3.

Reply:
Right, thanks for checking that. Now that I searched for it, I found it in the docs as well.
I have now added the checks